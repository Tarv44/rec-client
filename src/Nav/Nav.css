nav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    font-size: 24px;
    padding: .4em 0;
    width: 100%;
    background-color: #5C18F0;
}

nav a {
    color: white;
}

.nav-logo {
    margin-left: 5%;
    text-transform: uppercase;
    font-family:'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.nav-links {
    margin-right: 5%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    color: white;
}

.nav-links > :first-child {
    margin-right: 1em;
}

