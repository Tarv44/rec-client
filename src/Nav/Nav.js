import { NavLink } from 'react-router-dom';
import './Nav.css'

export default function Nav(props) {
    const links = Boolean(props.username) 
        ? <div className='nav-links'>
            <NavLink to={'/dashboard'}>{props.username}</NavLink>
            {/* <NavLink to={'/new-exchange'}>Create Exchange</NavLink> */}
        </div>
        : <div className='nav-links'>
            <NavLink to={'/signup'}>Signup</NavLink>
            <NavLink to={'/login'}>Login</NavLink>
        </div>
    return (
        <nav>
            <NavLink to={'/'} className="nav-logo">Rotation</NavLink>
            {links}
        </nav>
    )
}