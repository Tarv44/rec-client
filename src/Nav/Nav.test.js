import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Nav from './Nav'
import { BrowserRouter } from 'react-router-dom'


describe('----- Nav -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <Nav />
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <Nav />
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})