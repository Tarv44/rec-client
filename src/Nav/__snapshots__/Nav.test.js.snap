// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Nav ----- renders the UI as expected 1`] = `
<nav>
  <a
    aria-current="page"
    className="nav-logo active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Rotation
  </a>
  <div
    className="nav-links"
  >
    <a
      aria-current={null}
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
    <a
      aria-current={null}
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</nav>
`;
